src/globus_sdk/services/timer/data.py

```diff
@@ -0,0 +1,113 @@
+    the ``data`` argument in
+    :meth:`create_job <globus_sdk.TimerClient.create_job>`.
+
+    Timer operates through the `Globus Automate API
+    <https://docs.globus.org/globus-automation-services/>`_. Crucially, the
+    ``callback_url`` parameter should always be the URL used to run an action provider.
+    Currently the only supported action provider for this is for Transfer. Thus, users
+    should generally only use the :meth:`~from_transfer_data` method here. Any other
+    usage is meant for internal purposes; proceed with caution!
+
+
+    :param callback_url: URL for the action which the Timer job will use.
+    :type callback_url: str
+    :param callback_body: JSON data which Timer will send to the Action Provider on
+        each invocation
+    :param start: The datetime at which to start the Timer job.
+    :type start: datetime.datetime or str
+    :param interval: The interval at which the Timer job should recur.
+    :type interval: datetime.timedelta
+    :type callback_body: dict
+    :param name: A (not necessarily unique) name to identify this job in Timer
+    :type name: str, optional
+    :param stop_after: A date after which the Timer job will stop running
+    :type stop_after: datetime.datetime, optional
+    :param stop_after_n: A number of executions after which the Timer job will stop
+    :type stop_after_n: int
+    :param scope: The scope Timer will need in order to authorize against the Action
+        Provider specified in the callback URL.
+    :type scope: str, optional
+
+    .. automethodlist:: globus_sdk.TimerJob
+    """
+
+    def __init__(
+        self,
+        callback_url: str,
+        callback_body: Dict[str, Any],
+        start: Union[datetime.datetime, str],
+        interval: Union[datetime.timedelta, int, None],
+        *,
+        name: Optional[str] = None,
+        stop_after: Optional[datetime.datetime] = None,
+        stop_after_n: Optional[int] = None,
+        scope: Optional[str] = None,
+    ) -> None:
+        super().__init__()
+        self["callback_url"] = callback_url
+        self["callback_body"] = callback_body
+        if isinstance(start, datetime.datetime):
+            self["start"] = start.isoformat()
+        else:
+            self["start"] = start
+        if isinstance(interval, datetime.timedelta):
+            self["interval"] = int(interval.total_seconds())
+        else:
+            self["interval"] = interval
+        if name is not None:
+            self["name"] = name
+        if stop_after is not None:
+            self["stop_after"] = stop_after.isoformat()
+        if stop_after_n is not None:
+            self["stop_after_n"] = stop_after_n
+        if scope is not None:
+            self["scope"] = scope
+
+    @classmethod
+    def from_transfer_data(
+        cls,
+        transfer_data: TransferData,
+        start: Union[datetime.datetime, str],
+        interval: Union[datetime.timedelta, int, None],
+        **kwargs: Any,
```

Comment:
I think we should not use `**kwargs` and instead expand out the full list of args (exactly as in `__init__`).

It's somewhat annoying as a maintainer, but it has good effects on the type annotations because the signature for this method will then contain things like `name: str | None`. Without doing this, a caller can pass `name=0` here and it will pass `mypy`, but then behave in a bad way at runtime.

Reply:
Hah, I started with kwargs as a hack, and found pylint complained about the docstring (apparently it recently changed that `\**kwargs` is the correct way to write it for sphinx?), so I was distracted by my little kwargs win. Yeah, I agree expanding the args is preferable.